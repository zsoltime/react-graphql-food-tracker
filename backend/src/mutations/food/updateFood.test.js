'use strict';

const updateFood = require('./updateFood');

describe('udpateFood mutation', () => {
  it('updates the food item', async () => {
    const args = {
      id: 'validFoodId',
      calories: 596,
      carb: 11.6,
      fat: 46,
      protein: 29.6,
      sugar: 5.9,
    };
    const mockedContext = {
      db: {
        mutation: {
          updateFood: jest.fn().mockImplementation(() => Promise.resolve({
            ...args,
            brand: 'Meridian',
            name: 'Crunchy peanut butter',
            user: 'validUserId',
          })),
        },
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([{ id: 'validUserId' }])),
        },
      },
      request: { userId: 'validUserId' },
    };

    const food = await updateFood(null, args, mockedContext);

    expect(mockedContext.db.query.users).toBeCalled();
    expect(
      mockedContext.db.query.users.mock.calls[0]
    ).toMatchSnapshot();
    expect(mockedContext.db.mutation.updateFood).toBeCalled();
    expect(
      mockedContext.db.mutation.updateFood.mock.calls[0]
    ).toMatchSnapshot();
    expect(food).toMatchSnapshot();
  });

  it('throws error if user is not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      updateFood(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
  it('throws error if user does not have permission', () => {
    const args = { id: 'validFoodId' };
    const mockedContext = {
      db: {
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([])),
        },
      },
      request: { userId: 'validUserId' },
    };

    expect(
      updateFood(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
