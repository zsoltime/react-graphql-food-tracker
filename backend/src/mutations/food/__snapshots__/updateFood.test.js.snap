// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`udpateFood mutation throws error if user does not have permission 1`] = `"You don't have permission to update this food"`;

exports[`udpateFood mutation throws error if user is not signed in 1`] = `"You must be signed in to update foods"`;

exports[`udpateFood mutation updates the food item 1`] = `
Array [
  Object {
    "where": Object {
      "AND": Array [
        Object {
          "foods_some": Object {
            "id": "validFoodId",
          },
          "id": "validUserId",
        },
      ],
    },
  },
]
`;

exports[`udpateFood mutation updates the food item 2`] = `
Array [
  Object {
    "data": Object {
      "calories": 596,
      "carb": 11.6,
      "fat": 46,
      "protein": 29.6,
      "sugar": 5.9,
    },
    "where": Object {
      "id": "validFoodId",
    },
  },
]
`;

exports[`udpateFood mutation updates the food item 3`] = `
Object {
  "brand": "Meridian",
  "calories": 596,
  "carb": 11.6,
  "fat": 46,
  "id": "validFoodId",
  "name": "Crunchy peanut butter",
  "protein": 29.6,
  "sugar": 5.9,
  "user": "validUserId",
}
`;
